src/GraphQL/Types/AutoRegisteringObjectGraphType.cs

```diff
@@ -48,7 +61,45 @@ protected virtual IEnumerable<PropertyInfo> GetRegisteredProperties()
         {
             return typeof(TSourceType)
                 .GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                .Where(p => p.PropertyType.IsValueType || p.PropertyType == typeof(string));
+                .Where(p => IsEnabledForRegister(p.PropertyType, true));
+        }
+
+        private static bool IsEnabledForRegister(Type propertyType, bool firstCall)
+        {
+            if (propertyType == typeof(string)) return true;
+
+            if (propertyType.IsValueType) return true; // TODO: requires discussion: Nullable<T>, enums, any struct
```

Comment:
This behavior seems to me to be wrong. You can’t just take and allow to register all structures/enums.
While left as it was before.